operator/controllers/syncer/csi_node.go

```diff
@@ -0,0 +1,583 @@
+				Name:      podMountDir,
+				MountPath: s.driver.GetKubeletRootDirPath(),
+				// MountPropagation: &mountPropagationB,
+			},
+			{
+				Name:      hostDev,
+				MountPath: hostDevPath,
+			},
+			{
+				Name:      config.CSIConfigMap,
+				MountPath: config.ConfigMapPath,
+			},
+		}
+
+		for _, cluster := range s.driver.Spec.Clusters {
+			secret := cluster.Secrets
+			secretVolumeMount := corev1.VolumeMount{
+				Name:      secret,
+				MountPath: config.SecretsMountPath + secret}
+			volumeMounts = append(volumeMounts, secretVolumeMount)
+
+			//There is already an error mesaage, logged by operator if the cacert
+			//is missing in case of secureSslMode is set to true. The error is
+			//logged when k8s tries to create a volume for cacert.
+			isSecureSslMode := cluster.SecureSslMode
+			cacert := cluster.Cacert
+			if isSecureSslMode && len(cacert) != 0 {
+				cacertVolumeMount := corev1.VolumeMount{
+					Name:      cacert,
+					MountPath: config.CAcertMountPath + cacert}
+				volumeMounts = append(volumeMounts, cacertVolumeMount)
+			}
+		}
+		return volumeMounts
+
+	case nodeDriverRegistrarContainerName:
+		return []corev1.VolumeMount{
+			{
+				Name:      pluginDir,
+				MountPath: s.driver.GetSocketDir(),
+			},
+			{
+				Name:      registrationDir,
+				MountPath: registrationDirPath,
+			},
+		}
+
+	case nodeLivenessProbeContainerName:
+		return []corev1.VolumeMount{
+			{
+				Name:      pluginDir,
+				MountPath: s.driver.GetSocketDir(),
+			},
+		}
+	}
+	return nil
+}
+
+//ensureVolumes returns volumes for CSI driver pods.
+func (s *csiNodeSyncer) ensureVolumes() []corev1.Volume {
+	logger := csiLog.WithName("ensureVolumes")
+	volumes := []corev1.Volume{
+		k8sutil.EnsureVolume(pluginDir, k8sutil.EnsureHostPathVolumeSource(s.driver.GetSocketDir(), "DirectoryOrCreate")),
+		k8sutil.EnsureVolume(registrationDir, k8sutil.EnsureHostPathVolumeSource(s.driver.GetKubeletRootDirPath()+config.PluginsRegistry, "Directory")),
+		k8sutil.EnsureVolume(podMountDir, k8sutil.EnsureHostPathVolumeSource(s.driver.GetKubeletRootDirPath(), "Directory")),
+		k8sutil.EnsureVolume(hostDev, k8sutil.EnsureHostPathVolumeSource(hostDevPath, "Directory")),
+		k8sutil.EnsureVolume(config.CSIConfigMap, k8sutil.EnsureConfigMapVolumeSource(config.CSIConfigMap)),
+	}
+
+	for _, cluster := range s.driver.Spec.Clusters {
+		volume := k8sutil.EnsureVolume(cluster.Secrets,
+			ensureSecretVolumeSource(cluster.Secrets))
+		volumes = append(volumes, volume)
+
+		// To enable SSL, both secureSslMode and cacert have to be passed in
+		// CSI CR manifest file.
+		// TODO: See if this validation can be done by CRD itself.
+		isSecureSslMode := cluster.SecureSslMode
+		if isSecureSslMode {
+			cacert := cluster.Cacert
+			if len(cacert) == 0 {
+				logger.Error(errors.New("SecureSslMode Error"),
+					"When secureSslMode is set to true, cacert has to be "+
+						"specified in CSI CR manifest file!")
+				logger.Info("The Communication with GPFS cluster with ID " +
```

Comment:
Checked on this, both secureSslMode and cacert are needed to be specified in CR, otherwise the communication with the cluster does not continue in insecure mode, instead it fails and I see that driver pod stays in crashloopback. Will remove the Info log from here and will adjust the Error log to show better description.